managed/src/main/java/com/yugabyte/yw/controllers/CertificateController.java

```diff
@@ -125,77 +110,62 @@ public Result getRootCert(UUID customerUUID, UUID rootCA) {
+      throw new YWServiceException(BAD_REQUEST, "Invalid Customer UUID: " + customerUUID);
     }
     return ApiResponse.success(certs);
   }
 
   public Result get(UUID customerUUID, String label) {
-    CertificateInfo cert = CertificateInfo.get(label);
-    if (cert == null) {
-      return ApiResponse.error(BAD_REQUEST, "No Certificate with Label: " + label);
-    } else {
-      return ApiResponse.success(cert.uuid);
-    }
+    CertificateInfo cert = CertificateInfo.getOrBadRequest(label);
+    return ApiResponse.success(cert.uuid);
   }
 
   public Result delete(UUID customerUUID, UUID reqCertUUID) {
-    CertificateInfo certificate = CertificateInfo.get(reqCertUUID);
-    if (certificate == null) {
-      return ApiResponse.error(BAD_REQUEST, "Invalid certificate.");
-    }
+    CertificateInfo certificate = CertificateInfo.getOrBadRequest(reqCertUUID);
     if (!certificate.customerUUID.equals(customerUUID)) {
-      return ApiResponse.error(BAD_REQUEST, "Certificate doesn't belong to customer");
+      throw new YWServiceException(BAD_REQUEST, "Certificate doesn't belong to customer");
     }
     if (!certificate.getInUse()) {
       if (certificate.delete()) {
         Audit.createAuditEntry(ctx(), request());
         LOG.info("Successfully deleted the certificate:" + reqCertUUID);
         return ApiResponse.success();
       } else {
-        return ApiResponse.error(INTERNAL_SERVER_ERROR, "Unable to delete the Certificate");
+        throw new YWServiceException(INTERNAL_SERVER_ERROR, "Unable to delete the Certificate");
       }
     } else {
-      return ApiResponse.error(BAD_REQUEST, "The certificate is in use.");
+      throw new YWServiceException(BAD_REQUEST, "The certificate is in use.");
     }
   }
 
   public Result updateEmptyCustomCert(UUID customerUUID, UUID rootCA) {
-    Form<CertificateParams> formData = formFactory.form(CertificateParams.class)
-        .bindFromRequest();
-    if (formData.hasErrors()) {
-      return ApiResponse.error(BAD_REQUEST, formData.errorsAsJson());
-    }
-    if (Customer.get(customerUUID) == null) {
-      return ApiResponse.error(BAD_REQUEST, "Invalid Customer UUID: " + customerUUID);
-    }
-    CertificateInfo certificate = CertificateInfo.get(rootCA);
-    if (certificate == null) {
-      return ApiResponse.error(BAD_REQUEST, "Invalid Cert ID: " + rootCA);
-    }
+    Form<CertificateParams> formData = formFactory.getFormDataOrBadRequest(
+        CertificateParams.class);
+    Customer.getOrBadRequest(customerUUID);
+    CertificateInfo certificate = CertificateInfo.getOrBadRequest(rootCA);
     if (!certificate.customerUUID.equals(customerUUID)) {
-      return ApiResponse.error(BAD_REQUEST, "Certificate doesn't belong to customer");
+      throw new YWServiceException(BAD_REQUEST, "Certificate doesn't belong to customer");
     }
     if (certificate.certType == CertificateInfo.Type.SelfSigned) {
```

Comment:
move these two if blocks to:
private Certificate.checkEditable();

and call inside setCustomCertInfo